02_knight_and_princess/roman_kuprii/main.cpp

```diff
@@ -0,0 +1,166 @@
+#include <array>
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+using std::endl;
+using std::to_string;
+using std::vector;
+
+using EdgeId = int;
+using VertexId = int;
+
+constexpr int INVALID_ID = -1;
+const std::string JSON_GRAPH_FILENAME = "graph.json";
+
+struct Edge {
+  const EdgeId id = INVALID_ID;
+  const std::array<VertexId, 2> connected_vertices;
+
+  Edge(const VertexId& start, const VertexId& end, const EdgeId& _id)
+      : id(_id), connected_vertices({start, end}) {}
+
+  std::string to_json() const {
+    std::string res;
+    res = "{ \"id\": ";
+    res += to_string(id);
+    res += ", \"vertex_ids\": [";
+    res += to_string(connected_vertices[0]);
+    res += ", ";
+    res += to_string(connected_vertices[1]);
+    res += "] }";
+    return res;
+  }
+};
+
+struct Vertex {
+ public:
+  const VertexId id = INVALID_ID;
+
+  explicit Vertex(const VertexId& _id) : id(_id) {}
+
+  std::string to_json() const {
+    std::string res;
+    res = "{ \"id\": ";
+    res += to_string(id) + ", \"edge_ids\": [";
+    for (int n : edges_ids_) {
+      res += to_string(n);
+      res += ", ";
+    }
+    res.pop_back();
+    res.pop_back();
+    res += "] }";
+    return res;
+  }
+
+  void add_edge_id(const EdgeId& _id) { edges_ids_.push_back(_id); }
+
+ private:
+  std::vector<EdgeId> edges_ids_;
+};
+
+bool vertex_check(const std::vector<Vertex>& vertices, const VertexId& id) {
+  bool check = 0;
+  for (const auto& vert : vertices)
+    if (vert.id == id)
+      check = 1;
+  return check;
+}
+
+bool edge_connection_check(const std::vector<Edge>& edges,
+                           const VertexId& out_id,
+                           const VertexId& dest_id) {
+  bool check = 0;
+  for (const auto& edge : edges) {
+    if (edge.connected_vertices[0] == out_id &&
+        edge.connected_vertices[1] == dest_id)
+      check = 1;
+  }
+  return check;
+}
+
+class Graph {
+ public:
+  std::string to_json() const {
+    std::string res;
+    res = "{ \"vertices\": [ ";
+    for (const auto& v_it : vertices_) {
+      res += v_it.to_json();
+      res += ", ";
+    }
+    res.pop_back();
+    res.pop_back();
+    res += " ], \"edges\": [ ";
+    for (const auto& e_it : edges_) {
+      res += e_it.to_json();
+      res += ", ";
+    }
+    res.pop_back();
+    res.pop_back();
+    res += " ] }\n";
+    return res;
+  }
+
+  void add_vertex() { vertices_.emplace_back(vertices_.size()); }
+
+  void connect_vertices(const VertexId& out_id, const VertexId& dest_id) {
+    // check if vertices exist
+    assert(vertex_check(vertices_, out_id) == 1);
+    assert(vertex_check(vertices_, dest_id) == 1);
```

Comment:
Так достаточно, функция уже возвращает `bool`
```suggestion
    assert(vertex_check(vertices_, out_id));
    assert(vertex_check(vertices_, dest_id));
```